sound/soc/sof/sof-priv.h

```diff
@@ -54,11 +54,17 @@ extern int sof_core_debug;
 	(IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_ENABLE_DEBUGFS_CACHE) || \
 	 IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_IPC_FLOOD_TEST))
 
-/* DSP D0ix sub-state */
-enum sof_d0_substate {
-	SOF_DSP_D0I0 = 0,	/* DSP default D0 substate */
-	SOF_DSP_D0I3,		/* DSP D0i3(low power) substate*/
-};
+#define	SOF_DSP_STATE_BIT_D0		BIT(0) /* dsp is in D0 or D3 */
+#define	SOF_DSP_STATE_BIT_I3		BIT(1) /* dsp is in I3 or I0 */
+/* dsp power state is under updating */
+#define	SOF_DSP_STATE_BIT_UPDATE	BIT(2)
+
+#define	SOF_DSP_STATE_D3	0 /* dsp is in D3 */
+#define	SOF_DSP_STATE_D0	SOF_DSP_STATE_BIT_D0 /* dsp is in D0 */
+/* dsp is in D0I3(low power state) */
+#define	SOF_DSP_STATE_D0I3	(SOF_DSP_STATE_BIT_D0 | SOF_DSP_STATE_BIT_I3)
+/* dsp will update to either D0 or D0I3 state automatically */
+#define	SOF_DSP_STATE_AUTO	SOF_DSP_STATE_BIT_UPDATE
```

Comment:
@keyonjie this patch is just difficult to understand "ASoC: SOF: refine DSP power states". Care to break it down?